src/ray/common/memory_monitor.cc

```diff
@@ -111,41 +111,71 @@ std::tuple<int64_t, int64_t> MemoryMonitor::GetMemoryBytes() {
   return std::tuple(system_used_bytes, system_total_bytes);
 }
 
-int64_t MemoryMonitor::GetCGroupV1MemoryUsedBytes(const char *path) {
+int64_t MemoryMonitor::GetCGroupV1MemoryUsedBytes(const char *stat_path,
+                                                  const char *usage_path) {
+  // Cgroup available memory calculation:
+  //  `memory.usage_in_bytes` - `memory.stat.total_cache`
+  // This value is consistent with values `MemTotal` `MemAvailable` `MemFree` in `/proc/meminfo`
+  //  and they have relationship of:
+  //  - `memory.usage_in_bytes` == `MemTotal` - `MemFree`
+  //  - `memory.stat.total_cache` == `MemAvailable` - `MemFree`
+  //
+  // Explanation: What's this part `MemAvailable` - `MemFree` memory for and why
+  //   we should treat this part memory as "Not-in-used" ?
+  // Linux OS tries to fully use the whole physical memory size, so that in many
+  // cases we can observe that the system has very little `MemFree` size,
+  // but at the same time system might have a large `MemAvailable` size,
+  // then this part `MemAvailable - MemFree` size is borrowed by Linux OS
+  // to cache pages / buffers , BUT, once user process requests to allocate memory,
+  // and there is no sufficient free memory, OS will evict cache data out of this
+  // part memory and allocate them to user proces.
+  //
+  // Explanation 2: Why don't use the formular
+  //  `memory.stat.total_rss` - `memory.stat.inactive_file_bytes` to compute the
+  // in-used memory ?
+  // In my test the calculated number is much less than `MemTotal - MemAvailable`.
+  //  so the formular must miss something (but I haven't figured out which part is missed)
+  // But in my test,
+  //  cgroup `memory.usage_in_bytes` value perfectly matches (`MemTotal` - `MemFree`) value,
+  //  so I am sure that `memory.usage_in_bytes` value is correct.
+  //  and cgroup `memory.stat.total_cache` value also perfectly matches
+  //  (`MemAvailable` - `MemFree`) value,
+  // so that I am confident that this formular must be correct:
+  //   available_mem = `memory.usage_in_bytes` - `memory.stat.total_cache`
```

Comment:
NOTE: 

Current master Ray code calculation problems:
1. It counts `memory.stat.total_cache` as "in-used" memory, which causes the issue that it heavily underestimate available memory,
and it leads to Ray memory monitor misleadingly triggered to kill Ray tasks while there are lots of available memory.

2. Inaccurate formula, see ”Explanation 2“ in above comment.